modules/cnf-image-based-upgrade-troubleshooting.adoc

```diff
@@ -0,0 +1,289 @@
+
+* Stateroot that is no longer required
+* Precaching resources
+* OADP CRs
+* `ImageBasedUpgrade` CR
+
+If the {lcao} fails to perform the above steps, it transitions to the `AbortFailed` or `FinalizeFailed` states.
+The condition message and log show which steps failed.
+
+.Example error message
+[source,yaml]
+----
+message: failed to delete all the backup CRs. Perform cleanup manually then add 'lca.openshift.io/manual-cleanup-done' annotation to ibu CR to transition back to Idle
+      observedGeneration: 5
+      reason: AbortFailed
+      status: "False"
+      type: Idle
+----
+
+Resolution::
+
+. Inspect the logs to determine why the failure occurred.
+
+. To prompt {lcao} to retry the cleanup, add the `lca.openshift.io/manual-cleanup-done` annotation to the `ImageBasedUpgrade` CR.
+
+After observing this annotation, {lcao} retries the cleanup and if it is successful, the `ImageBasedUpgrade` stage transitions to `Idle`.
+
+If the cleanup fails again, you can manually clean up the resources.
+
+[id="ztp-image-based-upgrade-troubleshooting-stateroot_{context}"]
+=== Cleaning up stateroot manually
+
+Issue::
+
+Stopping at the `Prep` stage, {lcao} cleans up the new stateroot, and finalizing after a successful upgrade or a rollback, normally, {lcao} cleans up the old stateroot.
+If this step fails, it is recommended that you inspect the logs to determine why the failure occurred. 
+
+Resolution::
+
+. Check if there are any existing deployments in the stateroot by running the following command:
++
+[source,terminal]
+----
+$ ostree admin status
+----
+
+. If there are any, clean up the existing deployment:
++
+[source,terminal]
+----
+$ ostree admin undeploy <index_of_deployment> 
+----
+
+. After cleaning up all the deployments of the stateroot, wipe the stateroot directory:
++
+--
+[WARNING]
+====
+Ensure that the booted deployment is not in this stateroot.
+====
+
+[source,terminal]
+----
+$ stateroot="<stateroot_to_delete>"
+$ unshare -m /bin/sh -c "mount -o remount,rw /sysroot && rm -rf /sysroot/ostree/deploy/${stateroot}"
+----
+--
+
+[id="ztp-image-based-upgrade-troubleshooting-oadp-resources_{context}"]
+=== Cleaning up OADP resources manually
+
+Issue::
+
+This step can fail due to connection issues between {lcao} and the S3 backend. By restoring the connection and adding the `lca.openshift.io/manual-cleanup-done` annotation, the {lcao} can successfully cleanup backup resources.
+
+Resolution::
+
+. Check the backend connectivity by running the following command:
++
+--
+[source,terminal]
+----
+$ oc get backupstoragelocations.velero.io -n openshift-adp
+----
+
+.Example output
+[source,terminal]
+----
+NAME                          PHASE       LAST VALIDATED   AGE   DEFAULT
+dataprotectionapplication-1   Available   33s              8d    true
+----
+--
+
+. Remove all backup resources and then add the `lca.openshift.io/manual-cleanup-done` annotation to the `ImageBasedUpgrade` CR.
```

Comment:
Rendering off in the same way as above
![image](https://github.com/openshift/openshift-docs/assets/61474374/c567610d-8edb-4217-91ed-63a839bedf1c)

Reply:
Similar resolution to above but of course, not the same...